src/goose/console/server.clj

```diff
@@ -0,0 +1,42 @@
+(ns goose.console.server
+  (:require
+    [goose.brokers.redis.broker :as redis]
+    [goose.console.handler :as handler]
+    [ring.adapter.jetty :as jetty]
+    [ring.middleware.file :refer [wrap-file]]
+    [ring.middleware.resource :refer [wrap-resource]]
+    [ring.util.response :as response]
+    [goose.defaults :as d]))
+
+(defonce server (atom nil))
+
+(defn routes [{:keys [uri broker]}]
+  (if (= uri "/")
+    (handler/landing-page {:broker broker})
+    (response/response "<h1>Not Found</h1>")))
+
+(defn wrap-broker [handler broker]
+  (fn [req]
+    (handler (assoc req :broker broker))))
+
+(defn middleware [broker]
+  (-> routes
+      (wrap-broker broker)
+      (wrap-resource "public")
+      (wrap-file "resources/public")))
+
+(defn start-server [{:keys [broker
+                            server-config]}]
+  (reset! server
+          (jetty/run-jetty (middleware broker)
+                           (merge d/server-config server-config))))
+
+(defn stop-server []
+  (when-let [s @server]
+    (.stop s)
+    (reset! server nil)
+    (println "Server stopped")))
+
+(comment (start-server {:broker        (redis/new-producer redis/default-opts)
```

Comment:
question: I am not fully sure if this is a good way to inject broker details when the user starts console. (`start-server` should ideally take only server configuration).I could not come up with better alternatives. What are your thoughts about this?
@olttwa